pkg/controller/controller_instance.go

```diff
@@ -460,9 +460,31 @@ func (c *controller) pollInstance(serviceClass *v1alpha1.ServiceClass, servicePl
 
 	switch response.State {
 	case osb.StateInProgress:
-		// The way the worker keeps on requeueing is by returning an error, so
-		// we need to keep on polling.
-		// TODO(vaikas): Update the instance condition with progress message here?
+		// if the description is non-nil, then update the instance condition with it
+		if response.Description != nil {
+			// The way the worker keeps on requeueing is by returning an error, so
+			// we need to keep on polling.
+			clone, err := api.Scheme.DeepCopy(instance)
+			if err != nil {
+				return err
+			}
+			toUpdate := clone.(*v1alpha1.Instance)
+			toUpdate.Status.AsyncOpInProgress = true
+
+			reason := asyncProvisioningReason
+			message := fmt.Sprintf("%s (%s)", asyncProvisioningMessage, *response.Description)
+			if deleting {
+				reason = asyncDeprovisioningReason
+				message = fmt.Sprintf("%s (%s)", asyncProvisioningMessage, *response.Description)
```

Comment:
Is there something to add to the test to make sure it's getting the right reason in the update condition?